src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -239,7 +239,7 @@ internal class MemberSignatureComparer : IEqualityComparer<Symbol>
         public static readonly MemberSignatureComparer LambdaReturnInferenceCacheComparer = new MemberSignatureComparer(
             considerName: false,                // valid invoke is always called "Invoke"
             considerExplicitlyImplementedInterfaces: false,
-            considerReturnType: false,          // do not care
+            considerReturnType: true,          // to differentiate Task types
```

Comment:
This is a bit mysterious to me. How does task-like affect this cache?
Also, the comment above may need refreshing accordingly.